packages/flutter_tools/test/general.shard/flutter_manifest_test.dart

```diff
@@ -385,6 +385,21 @@ flutter:
       expect(flutterManifest.isPlugin, true);
       expect(flutterManifest.androidPackage, 'com.example');
     });
+    test('allows a multi-plat plugin declaration', () async {
+      const String manifest = '''
+name: test
+flutter:
+    plugin:
+      platforms:
```

Comment:
Please add tests that cover the cases where these maps are malformed, esp as noted above where they are not maps.